pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -97,48 +116,51 @@ type CiliumBGPNeighbor struct {
 	// Its value defines the Time To Live (TTL) value used in BGP packets sent to the neighbor.
-	// When empty or zero, eBGP multi-hop feature is disabled. The value is ignored for iBGP peers.
+	// The value 1 implies that eBGP multi-hop feature is disabled. This field is ignored for iBGP peers.
 	//
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=255
-	EBGPMultihopTTL int `json:"eBGPMultihopTTL,omitempty"`
-	// ConnectRetryTime defines the initial value for the BGP ConnectRetryTimer (RFC 4271, Section 8).
-	// The default value for the ConnectRetryTime (if empty or zero) is 120 seconds.
-	// Rounded internally to the nearest whole second.
+	// +kubebuilder:default=1
+	EBGPMultihopTTL *int32 `json:"eBGPMultihopTTL,omitempty"`
+	// ConnectRetryTimeSeconds defines the initial value for the BGP ConnectRetryTimer (RFC 4271, Section 8).
 	//
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:validation:Format=duration
-	ConnectRetryTime metav1.Duration `json:"connectRetryTime,omitempty"`
-	// HoldTime defines the initial value for the BGP HoldTimer (RFC 4271, Section 4.2).
-	// The default value for the HoldTime (if empty or zero) is 90 seconds.
-	// Rounded internally to the nearest whole second. Updating this value will cause a session reset.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=2147483647
+	// +kubebuilder:default=120
+	ConnectRetryTimeSeconds *int32 `json:"connectRetryTimeSeconds,omitempty"`
+	// HoldTimeSeconds defines the initial value for the BGP HoldTimer (RFC 4271, Section 4.2).
+	// Updating this value will cause a session reset.
 	//
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:validation:Format=duration
-	HoldTime metav1.Duration `json:"holdTime,omitempty"`
-	// KeepaliveTime defines the initial value for the BGP KeepaliveTimer (RFC 4271, Section 8).
-	// The default value for the KeepaliveTime (if empty or zero) is 1/3 of the HoldTime.
-	// Rounded internally to the nearest whole second. Updating this value will cause a session reset.
+	// +kubebuilder:validation:Minimum=3
+	// +kubebuilder:validation:Maximum=2147483647
+	// +kubebuilder:default=90
+	HoldTimeSeconds *int32 `json:"holdTimeSeconds,omitempty"`
+	// KeepaliveTimeSeconds defines the initial value for the BGP KeepaliveTimer (RFC 4271, Section 8).
+	// It can not be larger than HoldTimeSeconds. Updating this value will cause a session reset.
 	//
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:validation:Format=duration
-	KeepAliveTime metav1.Duration `json:"keepAliveTime,omitempty"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=2147483647
+	// +kubebuilder:default=30
+	KeepAliveTimeSeconds *int32 `json:"keepAliveTimeSeconds,omitempty"`
 	// GracefulRestart defines graceful restart parameters which are negotiated
 	// with this neighbor.
```

Comment:
I'd suggest adding in something here about how Graceful Restart is disabled if this is empty or nil (which seems to be the case AFAICT).

This does mean that:
```yaml
    gracefulRestart:
      enabled: true
```
is the simplest way to enable this on input, which is a little verbose, but given that it means the timeout values will then be included in the output, I think it's well worth it.